src/odemis/gui/cont/acquisition.py

```diff
@@ -696,6 +694,31 @@ def _on_export_data_done(self, future):
         data = future.result()
         self._display_acquired_data(data)
 
+    def _add_feature_info_filename(self):
+        """
+        Add feature name, feature status and the counter at the end of the filename.
+        """
+        local_tab = self._tab_data.main.getTabByName("cryosecom-localization")
+        local_tab.tab_data_model.select_current_position_feature()
+        feature_name = local_tab.tab_data_model.main.currentFeature.value.name.value
+        feature_status = local_tab.tab_data_model.main.currentFeature.value.status.value
+        feature_pattern = f"{feature_name}-{feature_status}"
+
+        # When a new feature pattern is acquired, counter resets to zero
+        # If the feature pattern already exists, the counter is incremented/updated
+        # during the creation of filename
+        self._config.fn_count = '0'
+
+        if 'Feature' in self._config.fn_ptn:
+            self._config.fn_ptn = re.sub(r'(Feature).*', feature_pattern + '-{cnt}', self._config.fn_ptn)
+        else:
+            self._config.fn_ptn = re.sub('-{cnt}', '-' + feature_pattern + '-{cnt}', self._config.fn_ptn)
+
+        # update the filename for the current acquisition
+        self._filename.value = create_filename(self._config.pj_last_path, self._config.fn_ptn,
```

Comment:
I don't think that's the right abstraction model. You are fighting against `create_filename()` to make something that matches your expectations. I think you should keep it more straightforward: take the current filename as it is, add the "feature name + -001" at the end of it (and before the extension), and call `make_unique()` with the list of files in the directory. (or just compute the first count which is not existing yourself. 
=> No need to touch `util.filename`.